openstack/containerinfra/v1/clustertemplates/requests.go

```diff
@@ -0,0 +1,63 @@
+package clustertemplates
+
+import (
+	"net/http"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// CreateOptsBuilder Builder.
+type CreateOptsBuilder interface {
+	ToClusterCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts params
+type CreateOpts struct {
```

Comment:
Where in the Magnum API code are these fields defined?

Reply:
Basically here https://github.com/openstack/magnum/blob/stable/queens/magnum/api/controllers/v1/cluster_template.py#L36 but some of them are generated by Magnum not user input.